### PR TITLE
Add bashtop v0.9.25

### PR DESCRIPTION
**Pull request by tretinha**
## Description of changes
This adds bashtop, a resources monitor: https://github.com/aristocratos/bashtop

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by IogaMaster — CHANGES_REQUESTED** [review]
Thanks for contributing!
The commit should be titled, `bashtop: init at v0.9.25`

File: pkgs/tools/system/bashtop/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/aristocratos/bashtop";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.tretinha ];
+  };
+
+  installFlags = [ "PREFIX=${placeholder "out"}" ];
+
+  src = fetchFromGitHub {
```
Comment (IogaMaster) [inline pkgs/tools/system/bashtop/default.nix:16]: Swap where meta and src are, its a convention.

Structure:
- pname
- version
- src
- phases and extra settings about phases
- meta

Comment (tretinha) [reply pkgs/tools/system/bashtop/default.nix:7]: Hey, thank you for your review. I think I added a wrong commit here. I'll close it and start from a clean slate.

**Comment by IogaMaster** [comment]
Also, does it build a package.
As well as add it to top level packages.

**Comment by Gerg-L** [comment]
> The commit should be titled, `bashtop: init at v0.9.25`

It should be `bashtop: init at 0.9.25`

`v` should always be removed from versions

**Comment by IogaMaster** [comment]
I didn't know that, thanks gerg!

**Comment by tretinha** [comment]
[IogaMaster](https://github.com/IogaMaster): I built it locally with
 ```
 nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'
 ```
 Generating:
 ```
 $ tree /nix/store/009hrdqnxdadbzjwyikj8dw5pq0m2nf4-bashtop-0.9.25
/nix/store/009hrdqnxdadbzjwyikj8dw5pq0m2nf4-bashtop-0.9.25
├── bin
│   └── bashtop
└── share
    └── doc
        └── bashtop
            └── README.md

5 directories, 2 files
```

Is that enough or should I build it differently? Thanks!

**Comment by IogaMaster** [comment]
You can call it from nixpkgs though, you need to add it to top level/all package

**Comment by Gerg-L** [comment]
it should be put in `by-name`
https://github.com/NixOS/nixpkgs/blob/2c732a9b5a5a60d91c685c92b87db5b8f5cf5812/pkgs/by-name/README.md#example

**Comment by tretinha** [comment]
@Gerg-L @IogaMaster, updated it to be at `pkgs/by-name/ba/bashtop/package.nix`

**Comment by Gerg-L** [comment]
you've got to add yourself to the maintainers list
do so in a separate commit before the bashtop commit in this PR

**Comment by Gerg-L** [comment]
oh i just looked at the source repo... bashtop hasn't had any commits in 2 years and has been replaced with btop.

**Comment by wyndon** [comment]
> oh i just looked at the source repo... bashtop hasn't had any commits in 2 years and has been replaced with btop.

This. I don't see the point in adding bashtop which is unmaintained and replaced by btop. bpytop was removed for this exact reason : https://github.com/NixOS/nixpkgs/pull/216878

**Comment by eclairevoyant** [comment]
Closing as the upstream project is unmaintained and has long been superseded.

**Pull request closed without merging**